pkg/receive/multitsdb.go

```diff
@@ -32,7 +34,10 @@ import (
 	"github.com/thanos-io/thanos/pkg/store/labelpb"
 )
 
-type GetStatsFunc func(tenantID, statsByLabelName string) *tsdb.Stats
+type TSDBStats interface {
+	SingleTenantStats(tenantID, statsByLabelName string) *status.TenantStats
+	AllTenantStats(statsByLabelName string) []status.TenantStats
```

Comment:
Yes, the idea was to use nil to signify a missing tenant. If we merge the two methods into one, how would we request stats for only one tenant? Would we set tenantID to an empty string?

Reply:
If the reason is only to check whether the tenant exists or not we can definitely have another return value.
If later we want to implement `TSDBStats` for other purposes then we will find ourselves implementing two methods, but for the same purpose.

```go
type TSDBStats interface {
  TenantStats(statsByLabelName string, tenantIDs ...string) ([]status.TenantStats, error)
}
```

Since we might need to get only some tenant's stats, not all tenants. Using an array is better for me.

Reply:
Got it, I think it's a very good point. I unified both methods under one.